crates/prover/src/components/store_imm.rs

```diff
@@ -0,0 +1,354 @@
+            .enumerate()
+        {
+            let denom_0: PackedQM31 = memory_relation.combine(value0);
+            let mult_0: PackedQM31 = -PackedQM31::from(enabler_col.packed_at(i));
+            let denom_1: PackedQM31 = memory_relation.combine(value1);
+            let mult_1: PackedQM31 = PackedQM31::from(enabler_col.packed_at(i));
+
+            col2.write_frac(i, mult_0 * denom_0 + mult_1 * denom_1, denom_0 * denom_1);
+        }
+        col2.finalize_col();
+
+        let mut col3 = interaction_trace.new_col();
+        for (i, (value0, value1)) in claim_data.lookup_data.range_check_20[0]
+            .iter()
+            .zip(&claim_data.lookup_data.range_check_20[1])
+            .enumerate()
+        {
+            let denom_0: PackedQM31 = range_check_20_relation.combine(value0);
+            let denom_1: PackedQM31 = range_check_20_relation.combine(value1);
+
+            col3.write_frac(
+                i,
+                PackedQM31::from(enabler_col.packed_at(i)) * (denom_0 + denom_1),
+                denom_0 * denom_1,
+            );
+        }
+        col3.finalize_col();
+
+        let (trace, claimed_sum) = interaction_trace.finalize_last();
+        (trace, Self { claimed_sum })
+    }
+}
+
+pub struct Eval {
+    pub claim: Claim,
+    pub memory: relations::Memory,
+    pub registers: relations::Registers,
+    pub range_check_20: relations::RangeCheck_20,
+}
+
+impl FrameworkEval for Eval {
+    fn log_size(&self) -> u32 {
+        self.claim.log_size
+    }
+
+    fn max_constraint_log_degree_bound(&self) -> u32 {
+        self.log_size() + 1
+    }
+
+    #[allow(non_snake_case)]
+    fn evaluate<E: EvalAtRow>(&self, mut eval: E) -> E {
+        let one = E::F::from(M31::from(1));
+        let expect_opcode_id = E::F::from(M31::from(Opcode::StoreImm as u32));
+
+        // 11 columns
+        let enabler = eval.next_trace_mask();
+        let pc = eval.next_trace_mask();
+        let fp = eval.next_trace_mask();
+        let opcode_id = eval.next_trace_mask();
+        let off0 = eval.next_trace_mask();
+        let off1 = eval.next_trace_mask();
+        let off2 = eval.next_trace_mask();
+        let clock = eval.next_trace_mask();
+        let inst_prev_clock = eval.next_trace_mask();
+        let dst_prev_val = eval.next_trace_mask();
+        let dst_prev_clock = eval.next_trace_mask();
+
+        // Enabler is 1 or 0
+        eval.add_constraint(enabler.clone() * (one - enabler.clone()));
+
+        // Opcode is StoreImm
+        eval.add_constraint(enabler.clone() * (opcode_id.clone() - expect_opcode_id));
+
+        // Update registers
+        eval.add_to_relation(RelationEntry::new(
+            &self.registers,
+            -E::EF::from(enabler.clone()),
+            &[pc.clone(), fp.clone()],
+        ));
+        eval.add_to_relation(RelationEntry::new(
+            &self.registers,
+            E::EF::from(enabler.clone()),
+            &[pc.clone() + E::F::one(), fp.clone()],
```

Comment:
Why not use the variable one defined above ?
Not sure E::F::from(M31::from(1)) and E::F::one() are equivalent?